kore/src/Kore/Reachability/Claim.hs

```diff
@@ -584,8 +584,12 @@ checkImplicationWorker (ClaimPattern.refreshExistentials -> claimPattern) =
         -> Pattern RewritingVariableName
         -> CheckImplicationResult ClaimPattern
     examine AnyUnified { didAnyUnify } stuck
-      | not didAnyUnify = NotImplied claimPattern
-      | isBottom condition = Implied
+      | not didAnyUnify
+      , not (isBottom right) =
+          NotImplied claimPattern
+      | didAnyUnify
+      , isBottom condition =
+          Implied
```

Comment:
Let's reorder these cases now, because we know that we will need to in your other branch. That way, we can be sure that it works with the order switched.
```suggestion
      | didAnyUnify
      , isBottom condition =
          Implied
      | not didAnyUnify
      , not (isBottom right) =
          NotImplied claimPattern
```